### DISCUSSION TITLE
代码检查插件，比如json等文件的检查，确实有高亮，但是底层 问题panel 没有显示是为什么

### DISCUSSION THREAD

**Discussion by rose8928** [opensumi/core#1788] [Q&A]
我用的node14，electron环境，显示已经安装了插件，但是下面 问题panel 没有显示
![1](https://user-images.githubusercontent.com/35814595/195573759-00755ff3-9d2a-41f9-9e67-cc49109de0ab.png)
![2](https://user-images.githubusercontent.com/35814595/195573829-0359e046-65d8-4b54-bfb7-6dbfc3ca1c01.png)

**Comment by yantze** [comment]
还需要装 JSON Language Features (built-in)

**Comment by rose8928** [comment]
谢谢，我试一下！

**Reply by rose8928** [reply]
我把vscode所有内置插件都放到了自己项目里，js文件的字体颜色倒是能显示，但是还是不打印到问题panel里面。
另外我试了一下 JSON Language Features (built-in)，报错如下。
是为什么呢@yantze 
![屏幕截图 2022-11-16 184404](https://user-images.githubusercontent.com/35814595/202159841-3805f8b0-1d98-4067-8c73-055b39c3ea79.png)

**Reply by hacke2** [reply]
降级插件版本 1.60.2 试一下

**Reply by rose8928** [reply]
如何降级插件，需要自己找到github上的打包吗，还是哪里可以下载到不同该插件版本

**Reply by erha19** [reply]
@rose8928 你这里的报错问题与这里类似，https://github.com/opensumi/core/discussions/1885#discussioncomment-4025680 ，主要原因是插件版本要求的 engine 版本较新（>1.67.0）导致的报错，你可以参考上面的评论内容升级内核，也可以选择版本较老的插件，手动进行安装，也可以通过修改 download.ts 脚本去安装。

手动安装路径 https://open-vsx.org/extension/vscode/json -> 右侧选择版本 -> Download -> 修改下载包后缀从 .vsx 为 .zip -> 解压 -> 取出内部的 extensions 目录内容作为新的插件目录

**Reply by rose8928** [reply]
好的，谢谢，确实修改版本之后就可以了。另外我想问一下，通过修改download.ts的方式去安装，是哪里的download.ts文件呢，如何安装。

**Reply by erha19** [reply]
@rose8928 这里，指的是项目的安装脚本 https://github.com/opensumi/ide-startup/blob/master/scripts/download.js

**Reply by rose8928** [reply]
知道了，感谢

**Reply by rose8928** [reply]
@yantze ，再请问一下https://open-vsx.org/这个网站里的插件，有些和vscode的版本不对应，怎么看插件的哪个版本可以给vscode1.60.2使用

**Reply by yantze** [reply]
> @yantze ，再请问一下[https://open-vsx.org](https://open-vsx.org)  这个网站里的插件，有些和vscode的版本不对应，怎么看插件的哪个版本可以给vscode1.60.2使用

可以看插件里面的 package.json engine 字段，里面有指定 vscode 插件 api 的上限版本号：
https://github.com/redhat-developer/vscode-xml/blob/main/package.json#L45

**Reply by rose8928** [reply]
好的谢谢，还是要下载下来看，不支持在线看。

**Reply by rose8928** [reply]
打扰，我安装了git,git-base,git-ui但是首页clone repository是灰色，并且点击无反应。我的customVSCodeEngineVersion版本是1.60.2.
git版本是      {
        "name": "git",
        "version": "1.62.3"
      },
      {
        "name": "git-ui",
        "version": "1.55.2"
      },
      {
        "name": "git-base",
        "version": "1.64.0-next.68976705968"
      },
<img width="662" alt="git-fail" src="https://user-images.githubusercontent.com/35814595/203211528-f620a1a4-f057-4cd7-b6d7-98a37c60dc9f.png">

但是git模块打开是ok的
<img width="600" alt="git-ok" src="https://user-images.githubusercontent.com/35814595/203211643-491615a5-93b4-46f1-a523-5a1569d73a9c.png">

我以为版本不对，我或者修改了customVSCodeEngineVersion，或者适配相应的git，git-base版本，都无效，而且有些版本报错。
是为什么呢？git这个模块是可用的吗？

**Reply by erha19** [reply]
@rose8928 不同的问题建议开新的讨论区讨论，简单回复一下，Git 模块对于插件版本有比较严格的要求，不能随意组合，建议参考我们示例项目内的默认版本，同时，部分插件可能针对 VS Code 进行了定向适配，可能也存在一些兼容性问题，可以将本地运行时出现的问题提到新的讨论区里面，一起看下
